### ISSUE TITLE
db->tree cannot fetch singleton items via a union

### ISSUE THREAD

**Issue by awkay** [omcljs/om#515]
db->tree works fine on unions as long as the join point in the database is a collection of items. When joining on a singleton; however, it fails to retrieve the data.

I'll look through the code and submit a patch.

**Comment by awkay** [comment]
fixed in fb784f3

**Comment by levitanong** [comment]
I'm seeing this issue again.
Repro:
```clj
(defui Bar
  static om/IQuery
  (query [_]
    [:foo/type
     :foo/id
     :bar/tender]))

(defui Baz
  static om/IQuery
  (query [_]
    [:foo/type
     :foo/id
     :baz/ruten]))

(defui Foo
  static om/Ident
  (ident [_ {:keys [foo/type foo/id]}]
    [type id])
  static om/IQuery
  (query [_]
    {:foo.type/bar (om/get-query Bar)
     :foo.type/baz (om/get-query Baz)}))

(def sample-state
  {:foo.type/bar {0 {:foo/id 0
                     :foo/type :foo.type/bar
                     :bar/tender "Moe"}}
   :foo.type/baz {1 {:foo/id 1
                     :foo/type :foo.type/baz
                     :baz/ruten "MMA"}}})
```
REPL
```clj
(om/db->tree (om/get-query Foo) [:foo.type/bar 0] sample-state)
;; {}
(om/db->tree (om/get-query Foo) [[:foo.type/bar 0]] sample-state)
;; [{:foo/type :foo.type/bar, :foo/id 0, :bar/tender "Moe"}]
(om/db->tree (om/get-query Foo) [[:foo.type/bar 0] [:foo.type/baz 1]] sample-state)
;; [{:foo/type :foo.type/bar, :foo/id 0, :bar/tender "Moe"} {:foo/type :foo.type/baz, :foo/id 1, :baz/ruten "MMA"}]
```
